sympy/utilities/codegen.py

```diff
@@ -1833,10 +1833,13 @@ def get_code_generator(language, project, standard=None):
             language = 'C89'
         elif standard.lower() == 'c99':
             language = 'C99'
-    CodeGenClass = {"C": CCodeGen, "C89": C89CodeGen, "C99": C99CodeGen,
-                    "F95": FCodeGen, "JULIA": JuliaCodeGen,
-                    "OCTAVE": OctaveCodeGen,
-                    "RUST": RustCodeGen}.get(language.upper())
+    if isinstance(language, type):
+        CodeGenClass = language
```

Comment:
For the code gen class? Doesn't it just take the project name in `__init__`? I was originally hoping to be able to use an instance of a custom printer, but I didn't get that working because the code gen class instantiates the printer with different settings in a couple places (e.g. human, dereference, etc.).